cpp/include/cudf/table/row_operators.cuh

```diff
@@ -71,8 +128,15 @@ class element_equality_comparator {
         return false;
       }
     }
-    return lhs.element<Element>(lhs_element_index) ==
-           rhs.element<Element>(rhs_element_index);
+    // NaNs are equal
```

Comment:
I don't think we should have to pay for this code for non-floating-point types. I think there should be a different operator for floating-point types of Element using SFINAE.